src/lib/services/licences.js

```diff
@@ -131,6 +131,30 @@ const getLicenceAccountsByRefAndDate = async (documentRef, date) => {
 const flagForSupplementaryBilling = licenceId =>
   repos.licences.update(licenceId, { includeInSupplementaryBilling: INCLUDE_IN_SUPPLEMENTARY_BILLING.yes });
 
+const getLabelForInvoiceWithoutInvoiceNumber = invoice => {
```

Comment:
I feel like this logic might live as a getter on the model itself, e.g.
```
get displayLabel() {
  if(this.invoiceNumber) {
   return this.invoiceNumber;
  }
  // logic below...
}
```

It's necessary to override the `toJSON()` method in this case to include the above calculated property - this is done on a few of the other models.

Reply:
I thought this as well and tried it out, but I must've done something wrong because it wasn't working for me, but I will give it another go